src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/menu/AndroidShareMenu.java

```diff
@@ -0,0 +1,249 @@
+
+		int pages = Math.max(0, users.size() / 36);
+
+		// Draw background start
+		for (int i = 0; i < 9; i++) {
+			menu.addItem(i, new CustomItem(new ItemStack(Material.GRAY_STAINED_GLASS_PANE), " "));
+			menu.addMenuClickHandler(i, (pl, slot, item, cursor, action) -> false);
+		}
+
+		for (int i = 45; i < 54; i++) {
+			menu.addItem(i, new CustomItem(new ItemStack(Material.GRAY_STAINED_GLASS_PANE), " "));
+			menu.addMenuClickHandler(i, (pl, slot, item, cursor, action) -> false);
+		}
+		// Draw background end
+
+		// Add trusted player slot
+		menu.addItem(0, new CustomItem(HeadTexture.SCRIPT_UP.getAsItemStack(), SlimefunPlugin.getLocalization().getMessage("android.access-manager.menu.add-player-title"), SlimefunPlugin.getLocalization().getMessage("android.access-manager.menu.add-player")));
+		menu.addMenuClickHandler(
+				0,
+				(p1, slot, item, cursor, action) -> {
+					p1.closeInventory();
+
+					SlimefunPlugin.getLocalization().sendMessage(p1, "android.access-manager.messages.input");
+
+					ChatUtils.awaitInput(
+							p1,
+							message -> {
+								Player target = Bukkit.getPlayerExact(message);
+
+								if (target == null) {
+									SlimefunPlugin.getLocalization().sendMessage(p1, "android.access-manager.messages.cannot-find-player", msg -> msg.replace("%player%", p1.getName()));
+								} else {
+									addPlayer(p1, target, b, users);
+								}
+							});
+					return false;
+				});
+
+		// Display added trusted player(s)
+		if (!users.isEmpty()) {
+			List<String> displayUsers = users.subList(page, Math.min(users.size() - 1, perPageDisplay));
+
+			for (int index = 0; index < displayUsers.size(); index++) {
+				int slot = index + startSlot;
+				OfflinePlayer current = Bukkit.getOfflinePlayer(UUID.fromString(displayUsers.get(index)));
+				menu.addItem(
+						slot,
+						new CustomItem(SkullItem.fromPlayer(current), "&b" + current.getName(), SlimefunPlugin.getLocalization().getMessage("android.access-manager.menu.delete-player")));
+				menu.addMenuClickHandler(
+						slot,
+						(p1, slot1, item, cursor, action) -> {
+							if (action.isLeftClick()) {
+								removePlayer(p1, current, b, users);
+							}
+
+							return false;
+						});
+			}
+		}
+
+		if (pages > 0) {
+			menu.addItem(47, ChestMenuUtils.getPreviousButton(p, page, pages));
+			menu.addMenuClickHandler(
+					46,
+					(pl, slot, item, cursor, action) -> {
+						int next = page - 1;
+						if (next < 1) next = pages;
+						if (next != page) {
+							openShareMenu(p, b, next);
+						}
+						return false;
+					});
+
+			menu.addItem(51, ChestMenuUtils.getNextButton(p, page, pages));
+			menu.addMenuClickHandler(
+					50,
+					(pl, slot, item, cursor, action) -> {
+						int next = page + 1;
+						if (next > pages) {
+							next = 1;
+						}
+						if (next != page) {
+							openShareMenu(p, b, next);
+						}
+						return false;
+					});
+		}
+
+		menu.open(p);
+	}
+
+	private static void addPlayer(
```

Comment:
You can use annotation `@ParametersAreNonnullByDefault` instead of using `@Nonnull` for each variable